app/presenters/tree_builder_ops_rbac_menu.rb

```diff
@@ -0,0 +1,249 @@
+class TreeBuilderOpsRbacMenu < TreeBuilder
+  include CompressedIds
+
+  has_kids_for Hash, [:x_get_tree_hash_kids]
+
+  attr_reader :role, :features, :editable
+
+  def initialize(name, type, sandbox, build, role:, editable: false)
+    @role     = role
+    @editable = editable
+    @features = @role.miq_product_features.order(:identifier).pluck(:identifier)
+
+    super(name, type, sandbox, build)
+  end
+
+  private
+
+  def menu_tree
+    recurse_menu_items(filtered_menus, root_select)
+  end
+
+  def filtered_menus
+    menus = []
+
+    Menu::Manager.each do |section|
+      next if section.id == :cons && !Settings.product.consumption
+      next if section.name.nil?
+      next unless Vmdb::PermissionStores.instance.can?(section.id)
+
+      menus.push(section)
+    end
+
+    menus
+  end
+
+  def recurse_menu_items(items, checked = false)
+    items.map do |item|
+      case item
+      when Menu::Section
+        menu_section_to_node(item)
+      when Menu::Item
+        if item.feature.nil? || !MiqProductFeature.feature_exists?(item.feature)
+          next
+        end
+        menu_item_to_node(item, checked)
+      when String
+        feature_name_to_node(item, checked)
+      else
+        item
+      end
+    end
+  end
+
+  def menu_section_to_node(section)
+    node = default_node.merge!(
+      :id      => "#{node_id}___tab_#{section.id}",
+      :text    => _(section.name),
+      :tooltip => _("%{title} Main Tab") % {:title => section.name},
+      :data    => {}
+    )
+
+    node[:data][:kids] = recurse_menu_items(section.items).compact
+    select_if_kids_selected(node, node[:data][:kids])
+
+    node
+  end
+
+  def menu_item_to_node(item, checked = false)
+    details = MiqProductFeature.feature_details(item.feature)
+
+    node = default_node.merge!(
+      :id      => "#{node_id}__#{item.feature}",
+      :text    => _(details[:name]),
+      :tooltip => _(details[:description]) || _(details[:name]),
+      :select  => checked,
+      :data    => {}
+    )
+
+    node[:select] ||= features.include?(item.feature)
+
+    kids = MiqProductFeature.feature_children(item.feature)
+    node[:data][:kids] = recurse_menu_items(kids, node[:select])
+
+    select_if_kids_selected(node, node[:data][:kids])
+
+    node
+  end
+
+  def feature_name_to_node(feature, checked = false)
+    details = MiqProductFeature.feature_details(feature)
```

Comment:
If you had `MiqProductFeature` object in `feature` instead of a String. You wouldn't have to ask in db for details or children as it is all stored in the object. And you could avoid using Hash :)